KIC Upgrading (#1241)
Kong/developer.konghq.com: app/kubernetes-ingress-controller/faq/upgrading-gateway.md

---
title: Upgrading {{ site.base_gateway }} using Helm

description: |

>   What do I need to know when upgrading Kong Gateway on Kubernetes? DB-backed mode vs DB-less

Reviewer: ```suggestion
  What do I need to know when upgrading {{site.base_gateway}} on Kubernetes? DB-backed mode vs DB-less
```

content_type: reference
layout: reference

products:
  - kic

works_on:
  - on-prem
  - konnect

related_resources:
  - text: Upgrading {{ site.kic_product_name }}
    url: /kubernetes-ingress-controller/faq/upgrading-ingress-controller/
---

> Every {{ site.kic_product_name }} deployment consists of two components that can be upgraded independently (learn more in [deployment methods](/index/kubernetes-ingress-controller/#deployment-topologies)).

Reviewer: ```suggestion
Every {{ site.kic_product_name }} deployment consists of two components that can be upgraded independently:
```

> - {{ site.kic_product_name }} (a control plane),
> - {{ site.base_gateway }} (a data plane).

Reviewer: ```suggestion
- {{ site.kic_product_name }} (a Control Plane),
- {{ site.base_gateway }} (a Data Plane).
```

> To see the available {{ site.base_gateway }} images, see [kong/kong-gateway](https://hub.docker.com/r/kong/kong-gateway/tags) on Docker Hub:

Reviewer: ```suggestion
To see the available {{ site.base_gateway }} images, see [kong/kong-gateway](https://hub.docker.com/r/kong/kong-gateway/tags) on Docker Hub. For more information, see [deployment methods](/index/kubernetes-ingress-controller/#deployment-topologies).
```

## Prerequisites

> - {{ site.kic_product_name }} installed using the `kong/ingress` Helm chart.
> - Enure your Helm charts repository is up-to-date by running `helm repo update`.
> - [yq](https://github.com/mikefarah/yq) installed (for YAML processing).
> - Check the version of {{ site.base_gateway }} and {{ site.kic_product_name }} you're currently  running. 

Reviewer: ```suggestion
- {{ site.kic_product_name }} installed using the `kong/ingress` Helm chart
- Enure your Helm charts repository is up-to-date by running `helm repo update`
- [yq](https://github.com/mikefarah/yq) installed (for YAML processing)
- Check the version of {{ site.base_gateway }} and {{ site.kic_product_name }} you're currently  running:
```

   ```bash
   helm get values --all kong -n kong  | yq '{
     "gateway": .gateway.image.repository + ":" + .gateway.image.tag,
     "controller": .controller.ingressController.image.repository + ":" + .controller.ingressController.image.tag
   }'
   ```

   As an output, you should get the versions of {{ site.base_gateway }} and {{ site.kic_product_name }} deployed in your cluster, for example:

   ```text
   gateway: kong/kong-gateway:{{ site.data.gateway_latest.release }}
   controller: kong/kubernetes-ingress-controller:{{ site.data.kic_latest.release }}
   ```

    To understand which version of {{ site.base_gateway }} is compatible with your version of {{ site.kic_product_name }}, refer to the [compatibility matrix](/kubernetes-ingress-controller/version-compatibility/#kong-gateway).

{:.warning}
>  **Upgrading {{ site.base_gateway }} in DB mode**
>
> There may be database migrations to run when running {{ site.base_gateway }} in DB-backed mode.

> > Refer to [Upgrade {{ site.base_gateway }} 3.x.x](/gateway/upgrade/) to learn more about upgrade paths between different versions of {{ site.base_gateway }}.

Reviewer: ```suggestion
> See [Upgrade {{ site.base_gateway }} 3.x.x](/gateway/upgrade/) to learn more about upgrade paths between different versions of {{ site.base_gateway }}.
```

## Upgrade {{ site.base_gateway }} using Helm

1. Edit or create a `values.yaml` file so that it contains a `gateway.image.tag` entry. Set this value to the version of {{ site.base_gateway }} to be installed.

    ```yaml
    gateway:
      image:

>         tag: {{site.data.latest_gateway.release}}

Reviewer: ```suggestion
        tag: "{{site.data.latest_gateway.release}}"
```
This doesn't seem to be rendering, it was empty for me. The "" doesn't fix it locally, but I think it's still necessary since I got an error because it was expecting this to be a string instead of a float.
